From 573f8745ee96a4fb44bf515f84ed5bc3eff9339f Mon Sep 17 00:00:00 2001
From: Abhishek Dubey <abhi.dubey011999@gmail.com>
Date: Sun, 25 Aug 2019 19:27:58 +0530
Subject: [PATCH] newjackfix: Update: Updated blobs

1. wcd-mbhc-v2.c: Tweaked some vars and removed/commented conflicting code

2. defconf: Update: Added CONFIG_SND_JACK

3. soc-jack.c: Merged commits from shadow kernel
---
 arch/arm64/configs/lineageos_kenzo_defconfig |  1 +
 sound/soc/codecs/wcd-mbhc-v2.c               | 25 ++++++++++++--------
 sound/soc/soc-jack.c                         |  7 +-----
 3 files changed, 17 insertions(+), 16 deletions(-)

diff --git a/arch/arm64/configs/lineageos_kenzo_defconfig b/arch/arm64/configs/lineageos_kenzo_defconfig
index 43017203ff9..bd5df9df282 100644
--- a/arch/arm64/configs/lineageos_kenzo_defconfig
+++ b/arch/arm64/configs/lineageos_kenzo_defconfig
@@ -467,6 +467,7 @@ CONFIG_SND_SOC=y
 CONFIG_SND_SOC_MSM8X16=y
 CONFIG_SND_SOC_WCD_MBHC=y
 CONFIG_SND_SOC_QDSP6V2=y
+CONFIG_SND_JACK=y
 CONFIG_UHID=y
 CONFIG_HID_APPLE=y
 CONFIG_HID_ELECOM=y
diff --git a/sound/soc/codecs/wcd-mbhc-v2.c b/sound/soc/codecs/wcd-mbhc-v2.c
index 6680d469609..ed7fe217e05 100644
--- a/sound/soc/codecs/wcd-mbhc-v2.c
+++ b/sound/soc/codecs/wcd-mbhc-v2.c
@@ -57,21 +57,27 @@ module_param(det_extn_cable_en, int,
 MODULE_PARM_DESC(det_extn_cable_en, "enable/disable extn cable detect");
 
 /* AGNi Audio Jack Testing & Debuging (psndna88@gmail.com) */
-int hs_detect_plug_time_ms = (1 * 1000);
-int special_hs_detect_time_ms = (2 * 1000);
+int hs_detect_plug_time_ms = (1 * 500);
+int special_hs_detect_time_ms = (1 * 1000);
 int mbhc_button_press_threshold_min = 250;
 int wcd_fake_removal_min_period_ms = 100;
 int fake_rem_retry_attempts;
 int wcd_mbhc_spl_hs_cnt = 1;
 int wcd_mbhc_btn_press_compl_timeout_ms = 50;
 int max_imped = 60000;
-bool new_jack = true;
-
-static bool __init setup_new_jack(char *str)
-{
-	return new_jack;
-}
-__setup("android.audiojackmode=", setup_new_jack);
+bool new_jack = false;
+
+//static bool __init setup_new_jack(char *str)
+//{
+//	if (!strncmp(str, "stock", strlen(str))) {
+//		new_jack = false;
+//		fake_rem_retry_attempts = 3;
+//        pr_info("wcd-mbhc-v2: android.audiojackmode = STOCK..\n");
+//    }
+//
+//	return new_jack;
+//}
+//__setup("android.audiojackmode=", setup_new_jack);
 
 module_param_named(hs_detect_plug_time_ms, hs_detect_plug_time_ms, int, 0664);
 module_param_named(mbhc_button_press_threshold_min_ms, mbhc_button_press_threshold_min, int, 0664);
@@ -89,7 +95,6 @@ enum wcd_mbhc_cs_mb_en_flag {
 
 static bool skip_impdet_retry;
 static bool lineout_detected;
-static struct switch_dev accdet_data;
 
 static void wcd_mbhc_jack_report(struct wcd_mbhc *mbhc,
 				struct snd_soc_jack *jack, int status, int mask)
diff --git a/sound/soc/soc-jack.c b/sound/soc/soc-jack.c
index 9a0623d1459..b7973e494fa 100644
--- a/sound/soc/soc-jack.c
+++ b/sound/soc/soc-jack.c
@@ -65,7 +65,6 @@ void snd_soc_jack_report(struct snd_soc_jack *jack, int status, int mask)
 	struct snd_soc_codec *codec;
 	struct snd_soc_dapm_context *dapm;
 	struct snd_soc_jack_pin *pin;
-	unsigned int sync = 0;
 	int enable;
 
 	trace_snd_soc_jack_report(jack, mask, status);
@@ -93,16 +92,12 @@ void snd_soc_jack_report(struct snd_soc_jack *jack, int status, int mask)
 			snd_soc_dapm_enable_pin(dapm, pin->pin);
 		else
 			snd_soc_dapm_disable_pin(dapm, pin->pin);
-
-		/* we need to sync for this case only */
-		sync = 1;
 	}
 
 	/* Report before the DAPM sync to help users updating micbias status */
 	blocking_notifier_call_chain(&jack->notifier, jack->status, jack);
 
-	if (sync)
-		snd_soc_dapm_sync(dapm);
+	snd_soc_dapm_sync(dapm);
 
 	snd_jack_report(jack->jack, jack->status);
 
-- 
2.20.1

